commands/translate.py

```diff
@@ -0,0 +1,169 @@
+from commands.base import Command
+from discord import Embed
+from helpers import *
+from utils.embed_table import EmbedTable
+
+import googletrans
+import json
+import re
+
+HISTORY = 10
+HISTORY_LIMIT = 100
+FLAGS_FILE = "files/flags.json"
+AVATAR_FORMAT = "https://cdn.discordapp.com/avatars/{0.id}/{0.avatar}.png?size=128"
+TRANSLATE_COLOR = int('4885ed', 16)
+
+class Translate(Command):
+    desc = """Translates a marked message to another language (removes all emojis).
+    React with a flag and type `!translate` to translate the message to the corresponding language.
+    Optional `index` argument defines offset of message to translate in list of marked messages."""
+
+    async def eval(self, index=0):
+        if int(index) < 0:
+            raise CommandFailure('Invalid index!')
+
+        archive = await create_list(self.client.logs_from(self.message.channel,
+                                                            limit=HISTORY_LIMIT))
+
+        if not archive:
+            raise CommandFailure('No tagged messages in the last %s found!' % HISTORY_LIMIT)
+
+        if int(index) >= len(archive):
+            raise CommandFailure('Invalid index!')
+
+        lang_list = archive[int(index)]
+        for entry in lang_list:
+            #translate contents
+            translator = googletrans.Translator()
+            text = removeEmojis(entry.content)
+            entry.content = translator.translate(text, dest=entry.language).text
+
+            channel = self.message.channel
+            header = ''
+            if type(entry.flag) is str:
+                footer = "Translation to %s requested by %s" % (entry.flag, self.name)
```

Comment:
Perhaps useful to mention what language it was translated from? If possible... (I assume the Translator() is detecting the original language)

Reply:
I can work out what language it is, but I don't have a way to bind it to the correct flag so I'd only be able to output the lang code, which I thing looks janky. Still think we should do it?